lib/spec/openapi/utils.js

```diff
@@ -319,9 +319,22 @@ function prepareOpenapiMethod (schema, ref, openapiObject) {
   return openapiMethod
 }
 
+function prepareOpenapiSchemas (schemas, ref) {
```

Comment:
Could you describe what you need to archive here?
I'm not sure to fully understand

With some testing this is not working if I put additional schemas in the bottom test:

```diff
    instance.addSchema({ $id: 'OrderItem', type: 'object', properties: { id: { type: 'integer' } } })
+    instance.addSchema({ $id: 'ProductItem', type: 'object', properties: { id: { type: 'integer' } } })
    instance.addSchema({ $id: 'Order', type: 'object', properties: { products: { type: 'array', items: { $ref: 'OrderItem' } } } })
    instance.post('/', { schema: { body: { $ref: 'Order' }, response: { 200: { $ref: 'Order' } } } }, () => {})
+    instance.post('/other', { schema: { body: { $ref: 'ProductItem' } } }, () => {})
```

Reply:
Hi Ecomm, I d'like to replace local `$ref` with local  openapi components schema (prefixed by `#/components/schemas/`) in generated openapi components schema.
With that change, openapi components will be able to reference other local components.

I wrote a simple test of my use case (too simple may be) and I pick the function from [cjsewell@1a816ee](https://github.com/cjsewell/fastify-swagger/commit/1a816ee39c9e9a9d6e0a8e73d4705e7f68110c0f) which seems to work well on my test.

I going to improve my test (with a more complex case, has you done). In try too investigate why the test hang when you add other schema definitions.
Do you have a better idea on that function implementation?

Thx

Reply:
I have investigated on hanging test  you where face to.
I seems to be cause by unfinished fastify-swagger initialization whereas fastify `fastify.ready()` is ok.
I have rewrote test using async/await to provide more readable tests and I never have this trouble anymore.

Hanging test behavior also exists on your test case without pull request feature.